tsfc/driver.py

```diff
@@ -43,7 +43,11 @@ def compile_form(form, prefix="form", parameters=None):
 
     assert isinstance(form, Form)
 
-    fd = ufl_utils.compute_form_data(form)
+    # determine if we're in complex mode; coffee mode does not support complex
+    complx = parameters and "scalar_type" in parameters and 'complex' in parameters["scalar_type"]
```

Comment:
Perhaps move the setting up of parameters from `compile_integral` to here, and then use:

```
complx = "complex" in parameters["scalar_type"]
```

Aside, here we're just checking `"complex"`, elsewhere we check `"double complex"`.